jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/CachedLayerAndName.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.builder.steps;
+
+import com.google.cloud.tools.jib.cache.CachedLayer;
+import javax.annotation.Nullable;
+
+/** Simple structure to hold a pair of {#link CachedLayer} and its string name. */
+class CachedLayerAndName {
+
+  private CachedLayer cachedLayer;
+  @Nullable private String name;
+
+  CachedLayerAndName(CachedLayer cachedLayer, @Nullable String name) {
+    this.cachedLayer = cachedLayer;
+    this.name = name;
+  }
+
+  CachedLayer getCachedLayer() {
+    return cachedLayer;
+  }
+
+  @Nullable
```

Comment:
I think if it's not escaping into the API surface, this might be okay? Since error prone's job is to work off these annotations? But have we generally moved off `@Nullable` in our code?

Reply:
I think I'm tolerant with that for this type of class that is awkwardly required in Java in a verbose way just to return multiple values from a method. I even wish I don't define getters.